src/apps/bm_devkit/pub_example/user_code/user_code.cpp

```diff
@@ -0,0 +1,52 @@
+#include "user_code.h"
+#include "bm_pubsub.h"
+#include "uptime.h"
+#include <stdint.h>
+#include <string.h>
+
+/*
+This application demonstrates how to publish data over bristlemouth via the bm_pub function.
+The application publishes a message to the topic "pubsub_example" every second.
+The message is a simple string that includes the uptime in milliseconds.
+To test this application, you will need to run the sub_example application on another node.
+*/
+
+// Publish at a rate of 1 Hz
+static constexpr uint32_t PUBLISH_PERIOD_MS = (1000);
+// The topic type is a identifier for the topic to encode different data types, we don't need to worry about this value for the tutorial, 
+// other than that it must be the same for the publisher and subscriber.
+static constexpr uint32_t EXAMPLE_PUBLSIH_TOPIC_TYPE = (1);
+// The topic version is a version number for the topic, in case things change we don't need to worry about this value for the tutorial,
+// other than that it must be the same for the publisher and subscriber.
+static constexpr uint32_t  EXAMPLE_PUBLISH_TOPIC_VERSION = (1);
+// This is the topic to publish to, the subscriber will need to know this topic to receive the data (see the application sub_example)
+static const char *const EXAMPLE_PUBLISH_TOPIC = "pubsub_example";
+
+
+void setup(void) {
+  /* USER ONE-TIME SETUP CODE GOES HERE */
+}
+
+void loop(void) {
+  /* USER LOOP CODE GOES HERE */
+  /// This section demonstrates a simple application to publish data over bristlemouth every second.
+  static uint32_t publishTimer = uptimeGetMs();
+  // Publish data every PUBLISH_PERIOD_MS milliseconds.
+  if (((uint32_t)uptimeGetMs() - publishTimer >= PUBLISH_PERIOD_MS)) {
```

Comment:
If we use `uint64_t` then we can avoid the cast and make the line a little simpler/cleaner. No strong opinion. Resolve as you see fit.

Reply:
Ah I see, potentially makes the sprintf line a little less familiar. Like I said, resolve as you see fit.